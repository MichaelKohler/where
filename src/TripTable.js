import React from 'react';
import PropTypes from 'prop-types'; // eslint-disable-line import/no-extraneous-dependencies
import TripTableEntry from './TripTableEntry';
import './scss/triptable.scss';

const TripTable = ({ trips }) => {
  return (
    <table className="trips">
      <thead>
        <tr>
          <th>From</th>
          <th>Until</th>
          <th>Destination</th>
          <th>Country</th>
          <th>Description</th>
          <th>Flights</th>
        </tr>
      </thead>
      <tbody>
        { trips.map((trip) => { return <TripTableEntry trip={trip} />; }) }
      </tbody>
    </table>
  );
};

TripTable.defaultProps = {
  trips: []
};

TripTable.propTypes = {
  trips: PropTypes.array,
};

export default TripTable;
