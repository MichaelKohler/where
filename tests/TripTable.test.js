import React from 'react';
import { render } from '@testing-library/react';

import TripTable from '../src/TripTable';

test('should correctly shallow render TripTable', () => {
  const trips = [{
    dateFrom: '2020-02-02',
  }, {
    dateFrom: '2020-02-03',
  }, {
    dateFrom: '2020-02-04',
  }, {
    dateFrom: '2020-02-05',
  }];
  const { container } = render(<TripTable trips={trips} />);
  expect(container.firstChild).toMatchSnapshot();
});
