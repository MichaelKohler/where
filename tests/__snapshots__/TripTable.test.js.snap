// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly shallow render TripTable 1`] = `
<table
  class="trips"
>
  <thead>
    <tr>
      <th>
        From
      </th>
      <th>
        Until
      </th>
      <th>
        Destination
      </th>
      <th>
        Country
      </th>
      <th>
        Description
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      class=""
    >
      <td>
        Sun Feb 02 2020
      </td>
      <td>
        Invalid Date
      </td>
      <td />
      <td />
      <td />
    </tr>
    <tr
      class=""
    >
      <td>
        Mon Feb 03 2020
      </td>
      <td>
        Invalid Date
      </td>
      <td />
      <td />
      <td />
    </tr>
    <tr
      class=""
    >
      <td>
        Tue Feb 04 2020
      </td>
      <td>
        Invalid Date
      </td>
      <td />
      <td />
      <td />
    </tr>
    <tr
      class=""
    >
      <td>
        Wed Feb 05 2020
      </td>
      <td>
        Invalid Date
      </td>
      <td />
      <td />
      <td />
    </tr>
  </tbody>
</table>
`;
